src/facilities-context/FacilitiesContext.tsx

```diff
@@ -238,15 +239,38 @@ export const FacilitiesProvider: React.FC<{ children: React.ReactNode }> = ({
   // update facilities when reference mapping changes
   useEffect(
     () => {
-      if (state.loading || isEmpty(facilityToReference)) return;
-      facilitiesActions.receiveFacilities(
-        dispatch,
-        facilitiesActions.buildCompositeFacilities(
+      if (
+        state.loading ||
+        isEmpty(facilityToReference) ||
+        isEmpty(state.facilities)
+      )
+        return;
+
+      async function updateFacilities() {
+        let referenceFacilities = state.referenceFacilities;
+        // If user just synced new reference facilities and is updating
+        // facilityToReference for the first time after initialization, then
+        // referenceFacilities will not be loaded yet.
+        if (isEmpty(referenceFacilities)) {
```

Comment:
since we have to fetch the reference facilites in order to display the `SyncNoUserFacilities` modal anyway, what if we just added them to the context at that point so that this object won't be empty when the facilities get saved? (I guess we would have to export an additional action from the `FacilitiesContext` to support that)

Reply:
Oh yeah, that makes sense...we are already doing the fetching in the modal, it just gets set to an internal state. That's a better flow, I'll update this.